### PR TITLE
Fix constant expression type check for negative literals

### PR DESCRIPTION
**Pull request by heshanpadmasiri**
## Purpose
$subject

## Summary by CodeRabbit

* **Bug Fixes**
  * Reduced redundant diagnostics for constant expressions; clearer single-report errors.

* **New Features**
  * Added many example programs showing constants (integers, floats, arrays, mappings) and their printed outputs, including edge-case float behaviours.

* **Tests**
  * Added numerous integration fixtures and updated the integration skip list to adjust covered cases; includes semantic-error fixtures for invalid constant initializers.

* **Refactor**
  * Internal constant-type handling updated to rely on semantic type information.

### CODE REVIEW

**Comment by warunalakshitha** [comment]
Lets fix the code rabbit comments

File: semantics/semantic_analyzer.go [hunk 1]
```diff
@@ -453,38 +453,64 @@ func (ca *constantAnalyzer) Visit(node ast.BLangNode) ast.Visitor {
...
+		bLangExpr := n.(ast.BLangExpression)
+		hasErrors := false
+		validateConstantExpr(ca.ctx(), bLangExpr, func(e ast.BLangExpression) {
+			ca.semanticErr("expression is not a constant expression", e.GetPosition())
+			hasErrors = true
+		})
+		if hasErrors {
+			return nil
+		}
+		if !analyzeExpression(ca, bLangExpr, ca.expectedType) {
```
Comment (warunalakshitha) [inline semantics/semantic_analyzer.go]: always false and never return null. May be missing test

File: semantics/semantic_analyzer.go [hunk 2]
```diff
@@ -453,38 +453,64 @@ func (ca *constantAnalyzer) Visit(node ast.BLangNode) ast.Visitor {
...
+	case *ast.BLangListConstructorExpr:
+		for _, member := range e.Exprs {
+			validateConstantExpr(ctx, member, onNonConst)
+		}
+	case *ast.BLangMappingConstructorExpr:
+		for _, field := range e.Fields {
+			if kv, ok := field.(*ast.BLangMappingKeyValueField); ok {
+				validateConstantExpr(ctx, kv.ValueExpr, onNonConst)
+			}
+		}
```
Comment (warunalakshitha) [inline semantics/semantic_analyzer.go:489]: These seems not covered by tests

File: semantics/semantic_analyzer.go [hunk 3]
```diff
@@ -453,38 +453,64 @@ func (ca *constantAnalyzer) Visit(node ast.BLangNode) ast.Visitor {
...
 }
 
+func validateConstantExpr(ctx *context.CompilerContext, expr ast.BLangExpression, onNonConst func(ast.BLangExpression)) {
+	switch e := expr.(type) {
+	case *ast.BLangLiteral, *ast.BLangNumericLiteral:
+		// always valid
+	case *ast.BLangSimpleVarRef:
+		sym := ctx.GetSymbol(e.Symbol())
+		if vs, ok := sym.(*model.ValueSymbol); ok && vs.IsConst() {
+			return
```
[the remark this answers is not in the document]
Comment (heshanpadmasiri) [reply semantics/semantic_analyzer.go:470]: This can never happen

File: semantics/semantic_analyzer.go (hunk 1 above)
Comment (heshanpadmasiri) [reply semantics/semantic_analyzer.go]: Fixed

File: semantics/semantic_analyzer.go (hunk 2 above)
Comment (heshanpadmasiri) [reply semantics/semantic_analyzer.go:489]: fixed

File: semantics/semantic_analyzer.go [hunk 4]
```diff
@@ -442,49 +442,66 @@ func (ca *constantAnalyzer) Visit(node ast.BLangNode) ast.Visitor {
...
-			} else {
-				ca.expectedType = exprTy
-			}
-		default:
-			ca.semanticErr("expression is not a constant expression", n.GetPosition())
-			return nil
+		} else {
+			ca.expectedType = exprTy
 		}
+		return nil
```
[the remark this answers is not in the document]
Comment (heshanpadmasiri) [reply semantics/semantic_analyzer.go:458]: fixed

[the remark this answers is not in the document]
Comment (heshanpadmasiri) [reply semantics/semantic_analyzer.go:458]: Type branch return nil so this never visit expressions defined in types

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Comment by warunalakshitha** [comment]
Lets sync the PR
